### PR TITLE
Remove dependencies.list

### PR DESCRIPTION
**Pull request by kraenhansen**
## What, How & Why?

This removes the `dependencies.list` file as it's not used by the SDK anymore.
We use the package.json to determine the version of the SDK instead.

This also removes a few scripts that were not really used by anyone after we migrated to GHA for releasing.

### CODE REVIEW

File: packages/realm/package.json [hunk 1]
```diff
@@ -196,6 +242,8 @@
     "module_path": "generated/ts",
     "host": "https://static.realm.io",
     "remote_path": "realm-js-prebuilds/{version}",
-    "napi_versions": [6]
+    "napi_versions": [
+      6
+    ]
```
Comment (kraenhansen) [inline packages/realm/package.json:247]: These format updates happened while testing the `npm version` command. I thought I might as well leave it in.

File: .github/workflows/prepare-release.yml [hunk 2]
```diff
@@ -30,16 +30,8 @@ jobs:
...
-          jq '.version="${{ steps.update-changelog.outputs.new-version }}"' package.json > package.tmp
-          mv package.tmp package.json
-
-      - name: Update package-lock.json
-        run: npm install --ignore-scripts
-
-      - name: Update dependencies.list
-        run: sed -i 's/^VERSION=.*/VERSION=${{ steps.update-changelog.outputs.new-version }}/' dependencies.list
+      - name: Update version
+        run: npm version ${{ steps.update-changelog.outputs.new-version }} --workspace realm
```
Comment (kraenhansen) [inline .github/workflows/prepare-release.yml:34]: Using the `npm version` command will automatically update the package-lock.json in the root.

File: .github/workflows/pr-realm-js.yml [hunk 3]
```diff
@@ -226,7 +226,7 @@ jobs:
         with:
           name: realm-js-prebuilds
           path: |
-            dependencies.list
+            README.md
```
Comment (kraenhansen) [inline .github/workflows/pr-realm-js.yml:229]: Depending on a different file, now that `dependencies.list` is going away.

File: scripts/build-android.js [hunk 4]
```diff
@@ -20,6 +20,9 @@ const commandLineArgs = require("command-line-args");
 const fs = require("fs-extra");
 const path = require("path");
 const exec = require("child_process").execFileSync;
+
+const { version } = require("realm/package.json");
```
Comment (kraenhansen) [inline scripts/build-android.js:24]: Reading the package version from `package.json` instead of `dependencies.list`.

**Review by takameyer — APPROVED** [review]
:shipit: Just a few remarks.  Love to see deleted code

**Comment by takameyer** [comment]
Is there some documentation referring to `dependencies.list` that should also be udpated?

**Pull request merged**